### PR TITLE
dockerfile builders template portability

### PR DESCRIPTION
**Pull request by davidgamero**
The current implementation doesn't allow the `languages` package to be imported for use outside draft as it relies on the `builders` embed in `/pkg/languages` which embeds the `/pkg/languages/builders` directory that is only populated after go:generate is executed through the makefile.

Since the draft makefile isn't executed when importing draft packages, the `/pkg/languages/builders` directory doesn't exist and the embed fails.

By creating a `builders` property on the `languages` struct and creating a public embed of the `/templates/builders` folder, 
- existing functionality is retained
- users can supply custom `builders` fs.FS to create their own language dockerFile generation templates
-  the `languages` package can be imported and used outside draft (without any changes to makefiles) as follows:
```
import "github.com/Azure/draft/templates"
import "github.com/Azure/draft/pkg/languages"
...
l := languages.CreateLanguagesFromEmbedFS(template.Dockerfiles, dest)

l.createDockerfileForLanguage(lang,inputs,templateWriter)
```

### CODE REVIEW

File: pkg/languages/languages.go [hunk 1]
```diff
@@ -98,16 +95,17 @@ func (l *Languages) PopulateConfigs() {
 	}
 }
 
-func CreateLanguages(dest string) *Languages {
+func CreateLanguages(builders embed.FS, dest string) *Languages {
```
Comment (imiller31) [inline pkg/languages/languages.go:98]: Maybe let's rename this to `CreateLanguagesFromEmbedFS`

File: templates/builders.go [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package templates
+
+import "embed"
+
+var (
+	//go:embed all:builders
```
Comment (imiller31) [inline templates/builders.go:6]: Maybe this is a good time for a rename from builders to Languages or Dockerfiles? Builders is left over from `draft-classic`

Comment (davidgamero) [reply templates/builders.go:6]: that's a great idea I was wondering why it was builders haha

File: pkg/languages/languages.go (hunk 1 above)
Comment (davidgamero) [reply pkg/languages/languages.go:98]: sounds good

**Review by bfoley13 — APPROVED** [review]

**Comment by imiller31** [comment]
LGTM merging

**Pull request merged**